### PR TITLE
Turn on GUC_REPORT flag for search_path to enable reporting back the parameter value upon change.

### PR DESCRIPTION
**Pull request by emelsimsek**
DESCRIPTION: Turns on the GUC_REPORT flag for search_path. This results in postgres to report the parameter status back in addition to Command Complete packet.

In response to the following command,

> SET search_path TO client1;

postgres sends back the following packets (shown in pseudo form):

C (Command Complete) SET  + **S (Parameter Status) search_path = client1**

### CODE REVIEW

File: src/backend/distributed/shared_library_init.c [hunk 1]
```diff
@@ -2602,6 +2602,17 @@ OverridePostgresConfigAssignHooks(void)
 			OldApplicationNameAssignHook = stringVar->assign_hook;
```
Comment (marcocitus) [inline src/backend/distributed/shared_library_init.c:2602]: maybe OverridePostgresConfigAssignHooks -> OverridePostgresConfigProperties

File: src/backend/distributed/shared_library_init.c [hunk 2]
```diff
@@ -2602,6 +2602,17 @@ OverridePostgresConfigAssignHooks(void)
...
 			stringVar->assign_hook = ApplicationNameAssignHook;
 		}
+
+		/*
+		 * Turn on GUC_REPORT for search_path. GUC_REPORT provides that an S (Parameter Status)
+		 * packet is appended after the C (Command Complete) packet sent from the server
+		 * for SET command. S packet contains the new value of the parameter
+		 * if its value has been changed.
+		 */
+		if (strcmp(var->name, "search_path") == 0)
```
Comment (JelteF) [inline src/backend/distributed/shared_library_init.c:2612]: Not introduced by this PR, but this whole function can be simplified a lot. There's no need to loop over the full list of GUCs. We can simply call `find_option` to fetch the desired GUC from the hashmap instead. The primary storage is actually a hashmap and the `get_guc_variables` function above allocates a completely new array.

Comment (JelteF) [reply src/backend/distributed/shared_library_init.c:2612]: Let's at least use `find_option` for `search_path`.  Then in a simple follow up PR we can do the refactor of the `application_name`.

Comment (emelsimsek) [reply src/backend/distributed/shared_library_init.c:2612]: > Let's at least use `find_option` for `search_path` (outside of this loop). Then in a simple follow up PR we can do the refactor of the `application_name`.

It looks like find_option is not exported by postgres.

Comment (JelteF) [reply src/backend/distributed/shared_library_init.c:2612]: Hmm you're right. It's only exported on the current master branch. So lets indeed keep the code as is. But lets add a comment that starting PG16 we can rewrite this using `find_option`.

File: src/backend/distributed/shared_library_init.c [hunk 3]
```diff
@@ -2602,6 +2602,17 @@ OverridePostgresConfigAssignHooks(void)
...
+		{
+			var->flags |= GUC_REPORT;
```
[the remark this answers is not in the document]
Comment (JelteF) [reply src/backend/distributed/shared_library_init.c:2614]: Yes this is fixed now: https://github.com/postgres/postgres/commit/2432b1a04087edc2fd9536c7c9aa4ca03fd1b363

First version it appeared in was PG14. So it's a good thing, we're dropping support for PG13 in the next version.

**Review by JelteF — APPROVED** [review]
Looks good to merge. The function rename that marco suggested sounds nice to do though: OverridePostgresConfigAssignHooks -> OverridePostgresConfigProperties

But not critical.

File: src/backend/distributed/shared_library_init.c (hunk 1 above)
Comment (marcocitus) [reply src/backend/distributed/shared_library_init.c:2602]: bump

**Pull request merged**